### ISSUE TITLE
Simulated input within Unity editor gets blocked by Sphere Collider child of Hand Controller

### ISSUE THREAD

**Issue by limepixl** [microsoft/MixedRealityToolkit-Unity#11321]
## Describe the bug

When developing inside the Unity editor and using the input simulation via mouse, the HandController instances block interaction between the user and any UI components that are "behind" the colliders of the HandControllers. More specifically, the collider that, when enabled, blocks the click/hover interactions, is the Sphere Collider that is part of the AttachTransform child of each Hand Controller.

## To reproduce

1. Clone this repo and checkout the following commit: a577eb290f (last **tested** commit that has the issue) (mrtk3 branch)
2. Open the MRTKDevTemplate project in Unity version 2021.3.16f1
3. Open the scene "CanvasExample"
4. Press Play
5. Do not move camera, hover over the UI components right in front of camera, and confirm that interaction happens on hover or click
6. Move backwards while interacting with same UI components. At some point the interaction will be blocked and UI components will not react to interaction attempts.
7. (Optional) Move forwards, left, right, above or below until interaction begins to be registered again.

## Expected behavior

It is expected to be able to interact with UI components in the scene using the CanvasProxyInteractor, looking from any angle, if in direct line of sight. The hand controllers are not visible, so the user would not know what might be blocking the interaction between the camera and the UI components.

## Screenshots

![image](https://user-images.githubusercontent.com/24658281/209166238-565f1958-2b94-43bb-9e00-f1ea708c2d6a.png)
![image](https://user-images.githubusercontent.com/24658281/209166578-31a2cc98-6f1b-4cb6-9137-0ce1eb30ae3d.png)

## My setup

- Unity 2021.3.16f1
- Using following MRTK3 package versions
<details>
<summary>MRTK3 package versions displayed in Unity</summary>

com.microsoft.mixedreality.openxr: 1.7.0
com.microsoft.mixedreality.visualprofiler: 2.2.0
com.microsoft.mrtk.accessibility: 1.0.0-development
com.microsoft.mrtk.audio: 3.0.0-development
com.microsoft.mrtk.core: 3.0.0-development
com.microsoft.mrtk.data: 1.0.0-development
com.microsoft.mrtk.diagnostics: 3.0.0-development
com.microsoft.mrtk.extendedassets: 3.0.0-development
com.microsoft.mrtk.graphicstools.unity: 0.4.0
com.microsoft.mrtk.input: 3.0.0-development
com.microsoft.mrtk.spatialmanipulation: 3.0.0-development
com.microsoft.mrtk.standardassets: 3.0.0-development
com.microsoft.mrtk.tools: 3.0.0-development
com.microsoft.mrtk.tts.windows: 1.0.0
com.microsoft.mrtk.uxcomponents: 3.0.0-development
com.microsoft.mrtk.uxcomponents.noncanvas: 3.0.0-development
com.microsoft.mrtk.uxcore: 3.0.0-development
com.microsoft.mrtk.windowsspeech: 3.0.0-development

</details>

## Target platform

This issue happens while developing inside the Unity editor on PC, regardless of the targeted platform.

**Comment by keveleigh** [comment]
Thanks for the report! I'm able to repro locally and will take a look.

**Comment by Zee2** [comment]
Ah, this is actually a known issue, first reported by @JPalmerDev a while ago. I don't think he or I ever got around to filing an issue for it. This should be fixable by either

- ensuring the hand colliders/triggers are properly disabled when the hand has "lost tracking" in the sim
- ensuring that the CanvasProxyInteractor raycast/layermask doesn't intersect whatever the layer of the hand colliders are
- or both

Assigning myself for now.

**Comment by keveleigh** [comment]
@Zee2 If you haven't already started, I have a branch going! The two culprits are the `GrabInteractor`'s collider and the `NearInteractionModeDetector`'s collider.

**Comment by keveleigh** [comment]
Fixed by #11328. I'll have a follow-up PR properly managing the lifecycle of these colliders too
